### ISSUE TITLE
Extend vector trick to matrix–matrix multiplication

### ISSUE THREAD

**Issue by wesselb** [MichielStock/Kronecker.jl#36]
If I'm not mistaken, the trick for matrix–vector multiplication can be extended to matrix–matrix multiplication, as follows:

```julia
reshape_cols(x, sizes...) = reshape(x, sizes...)
reshape_rows(x, sizes...) = transpose(reshape(transpose(x), reverse(sizes)...))

kron_id_a(a, x) = reshape_cols(a * reshape_cols(x, size(a)[2], :), :, size(x)[2])
kron_a_id(a, x) = reshape_rows(a * reshape_rows(x, size(a)[2], :), :, size(x)[2])
kron_a_b(a, b, x) = kron_id_a(b, kron_a_id(a, x))

a = randn(2, 2)
b = randn(3, 3)
x = randn(6, 6)

@test isapprox(kron(a, Diagonal(ones(3))) * x, kron_a_id(a, x))
@test isapprox(kron(Diagonal(ones(2)), b) * x, kron_id_a(b, x))
@test isapprox(kron(a, b) * x, kron_a_b(a, b, x))
```

This might be worthwhile implementing. Like the current implementation, care must be taken to choose the most efficient order: `kron_id_a(b, kron_a_id(a, x))` or `kron_a_id(a, kron_id_a(b, x))`.

**Comment by willtebbutt** [comment]
If I'm not mistaken, this is closed by #38 ?

**Comment by MichielStock** [comment]
Thanks @wesselb for the suggestion and @willtebbutt for the realization.
